Coop signaling reimplementation

vagechirkov opened the pull request:
Hi, I have started to reimplement cooperative signaling, based on the development branch.

This is still a work in progress, but I think the core approach to implementation should already be clear: I use inheritance to extend and override existing classes.

@mezdahun please let me know if you have any comments or ideas. I will continue to work on this PR and will notify you when it is ready to merge.

vagechirkov committed:
update gitignore with the common templates

vagechirkov committed:
create the app.py specific for the cooperative signaling project

vagechirkov committed:
make save_isims_env function more universal

vagechirkov committed:
isolate the playground startup function for the cooperative signaling

vagechirkov committed:
refactor the app.py file

vagechirkov committed:
add cooperation signaling simulation related classes

mezdahun commented:
Awesome!!! I will have a look tomorrow.

On Thu, Nov 3, 2022 at 4:22 PM Valerii ***@***.***> wrote:

> Hi, I have started to reimplement cooperative signaling, starting from the
> development branch.
>
> This is still a work in progress, but I think the core approach to
> implementation should already be clear: I use inheritance to extend and
> override existing classes.
>
> @mezdahun <https://github.com/mezdahun> please let me know if you have
> any comments or ideas. I will continue to work on this PR and will notify
> you when it is ready to merge.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/scioip34/ABM/pull/47
> Commit Summary
>
>    - a5b91f9
>    <https://github.com/scioip34/ABM/pull/47/commits/a5b91f9826acab852fd8c8ea55957107c1a0a748>
>    update gitignore with the common templates
>    - 63b38d2
>    <https://github.com/scioip34/ABM/pull/47/commits/63b38d2c353e10413fab4c6637e639e3ae79a580>
>    create the app.py specific for the cooperative signaling project
>    - 3f67e1d
>    <https://github.com/scioip34/ABM/pull/47/commits/3f67e1d7cd291dbaa7924e9a9b1d445d153a9012>
>    make save_isims_env function more universal
>    - 763cd1c
>    <https://github.com/scioip34/ABM/pull/47/commits/763cd1c9a6528d6df43fc497b8d278db084ea37e>
>    isolate the playground startup function for the cooperative signaling
>    - 6ce31b2
>    <https://github.com/scioip34/ABM/pull/47/commits/6ce31b2e369206d9f5e8aa61a9266e0c04e1e734>
>    refactor the app.py file
>    - 2e406fa
>    <https://github.com/scioip34/ABM/pull/47/commits/2e406fa49a7ca5c0ed821f65941cd10b4fbc7ae6>
>    add cooperation signaling simulation related classes
>
> File Changes
>
> (8 files <https://github.com/scioip34/ABM/pull/47/files>)
>
>    - *M* .gitignore
>    <https://github.com/scioip34/ABM/pull/47/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
>    (248)
>    - *M* abm/app.py
>    <https://github.com/scioip34/ABM/pull/47/files#diff-799c248ba3eb8913e0f366b6b6630f5a15e4e50bd6ced122434acbb894807e02>
>    (20)
>    - *A* abm/projects/__init__.py
>    <https://github.com/scioip34/ABM/pull/47/files#diff-3810d8ff7dc472c3809574532481e394e623cccf7f5c050b342ad89a502a3a19>
>    (0)
>    - *A* abm/projects/cooperative_signaling/README.md
>    <https://github.com/scioip34/ABM/pull/47/files#diff-5b233c5374a5f25eef2bdfb1d38a4d26fb6cd7f4528287e57b7fe68a307b4cf7>
>    (22)
>    - *A* abm/projects/cooperative_signaling/__init__.py
>    <https://github.com/scioip34/ABM/pull/47/files#diff-e66ec51816ef001bb9e1142af73417f3f1c75d5c007c547683dcc278738c002d>
>    (0)
>    - *A* abm/projects/cooperative_signaling/app.py
>    <https://github.com/scioip34/ABM/pull/47/files#diff-e4ca00c689443a77182219317e29cd2aac42dde7f1b8204a7b24aa79c00720e7>
>    (38)
>    - *A* abm/projects/cooperative_signaling/cooperative_signaling.env
>    <https://github.com/scioip34/ABM/pull/47/files#diff-bcde257265e07f9fe3fb51e69d82a6d25531882fa38834a090acf13f87ffde6b>
>    (101)
>    - *A* abm/projects/cooperative_signaling/simulation.py
>    <https://github.com/scioip34/ABM/pull/47/files#diff-d199c98ddfde1dbec680c1f9c7d4049f9c75dd37963350900941702d81bbed23>
>    (402)
>
> Patch Links:
>
>    - https://github.com/scioip34/ABM/pull/47.patch
>    - https://github.com/scioip34/ABM/pull/47.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/scioip34/ABM/pull/47>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJEY2KSKXN2UU52AIPMIRATWGPKDPANCNFSM6AAAAAARWIE4GU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

vagechirkov commented:
There is a tricky part where the `CooperativeSignalingPlaygroundSimulation` inherits from the `PlaygroundSimulation` and `CooperativeSignalingSimulation` (`abm/projects/cooperative_signaling/simulation.py` line 240). I am not hundred percent sure about the behind-the-scenes magic that enables hierarchical attribute inheritance, but it seems to be working fine.

This entry might be helpful, but I'm still wrapping my head around it:
https://docs.python.org/3/tutorial/classes.html#multiple-inheritance

mezdahun commented:
Do you maybe have some time tomorrow for a zoom meeting? You could quickly
guide me through the diffs in the PR. I have seen that there are quite some
changes regarding the .env files. This on one hand I find very nice because
it needs improvement for sure, on the other hand we need to take care that
it remains compatible with the docker/singularity framework we have on the
HPC cluster.

On Thu, Nov 3, 2022 at 4:32 PM Valerii ***@***.***> wrote:

> There is a tricky part where the CooperativeSignalingPlaygroundSimulation
> inherits from the PlaygroundSimulation and CooperativeSignalingSimulation.
> I am not hundred percent sure about the behind-the-scenes magic that
> enables hierarchical attribute inheritance, but it seems to be working fine.
>
> This entry might be helpful, but I'm still wrapping my head around it:
> https://docs.python.org/3/tutorial/classes.html#multiple-inheritance
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/scioip34/ABM/pull/47#issuecomment-1302290231>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJEY2KVLNCZYGENDXIIMPIDWGPLJJANCNFSM6AAAAAARWIE4GU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

vagechirkov committed:
refactor class names and create CSAgent class

mezdahun commented on abm/projects/cooperative_signaling/app.py:
```diff
@@ -0,0 +1,38 @@
+    sim.start()
+
+
+if __name__ == "__main__":
+    start_playground()
```
Instead of boilerplate I would prefer adding a proper entrypoint via setup.py so that we can store multiple relevant use cases in this project specific app.py

vagechirkov replied:
Let's update `setup.py` in the different PR :)

mezdahun commented on abm/app.py:
```diff
@@ -73,19 +73,19 @@ def start_headless():
 
-def save_isims_env():
+def save_isims_env(env_dir, _EXP_NAME, _pgt, _envconf):
     """translating a default parameters dictionary to an environment
     file and using env variable keys instead of class attribute names"""
```
In case we now have function parameters we should include these in the docstring

vagechirkov committed:
Update cooperative agent

mezdahun committed:
testing access to fork

mezdahun commented:
1.) finish PR
2.) refactor current main project into project folder
3.) keep .env specific to runs in the main folder
4.) keep template .env files in project folders

vagechirkov committed:
refactor cooperative signaling folder structure

vagechirkov committed:
fix import errors

vagechirkov committed:
add cooperative signaling resource

vagechirkov committed:
add coop signaling resource and agents movement

vagechirkov commented:
@mezdahun I think everything is ready but for some reason the resource is invisible 🤔 Could you check it, please?

vagechirkov commented:
I'll write some tests in the meantime 🙂

vagechirkov committed:
add random walk happy test

mezdahun committed:
Only update position if the user is not holding the patch with the cursor

mezdahun committed:
Added missing part of draw_update

of CSResource

mezdahun committed:
Generalizing separation of update and draw_update

mezdahun committed:
Spearation of update and draw_updatae methods in interactions

vagechirkov committed:
add pytest setup

mezdahun committed:
CSAgent inherits drawing method from Agent

as currently they look the same. We only need to override if we want to give different look for CSAgent

vagechirkov committed:
add pytest workflow

vagechirkov committed:
change the import

vagechirkov committed:
update github testing workflow

vagechirkov committed:
update github testing workflow

mezdahun committed:
Merge remote-tracking branch 'forkvalerii/coop-signaling-reimplementation' into coop-signaling-reimplementation

vagechirkov committed:
add flake8 into github workflow

mezdahun committed:
Merge remote-tracking branch 'forkvalerii/coop-signaling-reimplementation' into coop-signaling-reimplementation

mezdahun commented:
Changes to agent and resource drawing included now

vagechirkov commented:
> Changes to agent and resource drawing included now

Cool!!

vagechirkov commented:
I added one mock test and the testing workflow, see how it looks here: https://github.com/vagechirkov/ABM/actions/runs/3393553257

mezdahun commented:
Awesome!

vagechirkov commented:
LGTM! Let's merge? :)

mezdahun commented:
bypassing branch protection due to Docker CLI test fail when pulling from forks.

vagechirkov commented:
> bypassing branch protection due to Docker CLI test fail when pulling from forks.

I think this is because I didn't properly specify environment variables in my fork. I will check that. But all tests passed after the merge 🙂

mezdahun commented:
Yes it i because the automatic push to dockerhub that requires credentials set up from the repo owner.
